packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -101,6 +100,17 @@ export class AwsCliCompatible {
       }
     }
 
+    if (!region && this.isEc2) {
```

Comment:
I don't enjoy that this relies on `credentialChain()` having been called before `region()` is called.

I'd prefer you re-use `hasEc2Credentials()` here, and make `hasEc2Credentials()` cache its return value.

Also it's probably appropriate to rename `hasEc2Credentials() -> isEc2Instance()` then.